File: CS/OsmGeocodeSample/MainForm.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace OsmGeocodeSample {
    public partial class MainForm : Form {
        public MainForm() {
            InitializeComponent();
        }

        #region #ConfigureMapLayers
        private void OnFormLoad(object sender, EventArgs e) {
            mapControl.Layers.AddRange(new LayerBase[] {
                new ImageLayer {
                    DataProvider = new OpenStreetMapDataProvider() {
                        TileUriTemplate = "YOUR_TILE_SERVICE.com/{1}/{2}/{3}"
                    }
                },
                new InformationLayer {
                    DataProvider = new OsmGeocodeDataProvider()
                }
            });
        }
        #endregion #ConfigureMapLayers
    }
}

File: VB/OsmGeocodeSample/MainForm.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace OsmGeocodeSample
    Partial Public Class MainForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        #Region "#ConfigureMapLayers"
        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            mapControl.Layers.AddRange(New LayerBase() { _
                New ImageLayer With { _
                    .DataProvider = New OpenStreetMapDataProvider() With {.TileUriTemplate = "YOUR_TILE_SERVICE.com/{1}/{2}/{3}"} _
                }, _
                _
                New InformationLayer With {.DataProvider = New OsmGeocodeDataProvider()} _
            })
        End Sub
        #End Region ' #ConfigureMapLayers
    End Class
End Namespace
